src/main/scala/org/broadinstitute/hail/variant/VariantSampleMatrix.scala

```diff
@@ -719,28 +719,29 @@ class VariantSampleMatrix[T](val metadata: VariantMetadata,
           Some(mBc.value(queries.head))
       }
       (inserter(va, toIns), gs)
-    }.toOrderedRDD(_.locus),
+    }.toOrderedRDD[Locus],
       vaSignature = newSignature)
   }
 
   def annotateVariants(otherRDD: RDD[(Variant, Annotation)], newSignature: Type,
     inserter: Inserter): VariantSampleMatrix[T] = {
-    val newRDD = rdd.orderedLeftJoinDistinct(otherRDD.toOrderedRDD(_.locus))
+    val newRDD = rdd.orderedLeftJoinDistinct(otherRDD.toOrderedRDD)
       .mapValues { case ((va, gs), annotation) =>
         (inserter(va, annotation), gs)
-      }.toOrderedRDD(_.locus)
+      }.toOrderedRDD[Locus]
     copy(rdd = newRDD, vaSignature = newSignature)
   }
 
   def annotateLoci(lociRDD: RDD[(Locus, Annotation)], newSignature: Type, inserter: Inserter): VariantSampleMatrix[T] = {
+    import Locus.orderedKey
+
     val newRDD = rdd
-      .mapMonotonic({ case (v, vags) => (v.locus, (v, vags)) },
-        identity[Locus])
-      .orderedLeftJoinDistinct(lociRDD.toOrderedRDD(identity[Locus]))
+      .mapMonotonic(OrderedKeyFunction(_.locus), { case (v, vags) => (v, vags) })
+      .orderedLeftJoinDistinct(lociRDD.toOrderedRDD)
       .mapValues { case ((v, (va, gs)), annotation) => (v, (inserter(va, annotation), gs)) }
-      .toOrderedRDD[Locus](identity)
-      .mapMonotonic({ case (_, vvags) => vvags },
-        (v: Variant) => v.locus)
+      .toOrderedRDD[Locus]
+      // FIXME: this isn't actually monotonic?
+      // .mapMonotonic(OrderedKeyFunction(???), _._2._2)
```

Comment:
@cseed So... I think I've convinced myself this is actually monotonic (but not _strictly_ monotonic). But it doesn't really work with the `OrderedKeyFunction` interface we've designed. Not sure if `OrderedKeyFunction` is actually worth the trouble at this point.